protobuf_release.bzl

```diff
@@ -0,0 +1,34 @@
+"""
+Generates package naming variables for use with rules_pkg.
+"""
+
+load("@rules_pkg//:providers.bzl", "PackageVariablesInfo")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load(":protobuf_version.bzl", "PROTOBUF_VERSION")
+
+def _package_naming_impl(ctx):
+  values = {}
+  values["product_name"] = ctx.attr.product_name
+  values["version"] = PROTOBUF_VERSION
+
+  # infer from the current cpp toolchain.
+  toolchain = find_cpp_toolchain(ctx)
+  values["cpu"] = toolchain.cpu
+  
+  return PackageVariablesInfo(values = values)
+
+
+package_naming = rule(
+  implementation = _package_naming_impl,
+    attrs = {
+      "product_name": attr.string(
+          doc = "Placeholder for our final product name.",
+      ),
+      "version": attr.string(
+          doc = "Placeholder for our release version.",
```

Comment:
Do we have to have these placeholders? If so, can we rename or put comments to explain how they will get initialized?

Reply:
I can remove version actually, but we should leave product name so we can set it dynamically.